File: app/src/main/java/com/tondz/nguoimu/NguoiMuSDK.java
package com.tondz.nguoimu;

import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.view.Surface;

import java.util.List;

public class NguoiMuSDK {
    public native boolean loadModel(AssetManager assetManager, int yoloDetect, int faceDectector, int trafficLight, int camDiec, int money, int chuCai);

    public native boolean openCamera(int facing);

    public native boolean closeCamera();

    public native boolean setOutputWindow(Surface surface);

    public native List<String> getListResult();

    public native String getEmbedding();

    public native Bitmap getFaceAlign();

    public native String getEmbeddingFromPath(String path);

    public native Bitmap getFaceAlignFromPath(String path);

    public native String getLightTraffic();

    public native String getEmotion();

    public native String getDeaf();
    public native String getChuCai();

    public native List<String> getListMoneyResult();

    public native Bitmap getImage();

    static {
        System.loadLibrary("nguoimusdk");
    }
}

File: app/src/main/java/com/dongnai/nguoikhuyettat/NguoiMuSDK.kt
package com.dongnai.nguoikhuyettat

import android.content.res.AssetManager
import android.graphics.Bitmap
import android.view.Surface

class NguoiMuSDK {
    external fun loadModel(
        assetManager: AssetManager?,
        yoloDetect: Int,
        faceDectector: Int,
        trafficLight: Int,
        camDiec: Int,
        money: Int,
        chuCai: Int
    ): Boolean

    external fun openCamera(facing: Int): Boolean

    external fun closeCamera(): Boolean

    external fun setOutputWindow(surface: Surface?): Boolean

    val listResult: List<String?>?
        external get

    val embedding: String?
        external get

    val faceAlign: Bitmap?
        external get

    external fun getEmbeddingFromPath(path: String?): String?

    external fun getFaceAlignFromPath(path: String?): Bitmap?

    val lightTraffic: String?
        external get

    val emotion: String?
        external get

    val deaf: String?
        external get
    val chuCai: String?
        external get

    val listMoneyResult: List<String?>?
        external get

    val image: Bitmap?
        external get

    companion object {
        init {
            System.loadLibrary("nguoimusdk")
        }
    }
}
